presto-orc/src/main/java/com/facebook/presto/orc/metadata/OrcMetadataReader.java

```diff
@@ -388,8 +392,44 @@ private static BinaryStatistics toBinaryStatistics(OrcProto.BinaryStatistics bin
         return new BinaryStatistics(binaryStatistics.getSum());
     }
 
+    static Slice toSlice(ByteString value)
+    {
+        return Slices.wrappedBuffer(value.toByteArray());
+    }
+
+    @VisibleForTesting
+    static Slice maxStringTruncateAtFirstReplacementCharacter(Slice value)
+    {
+        if (value == null) {
+            return null;
+        }
+
+        int index = value.indexOf(REPLACEMENT_CHARACTER_UTF8);
+        if (index == -1) {
+            return value;
+        }
+        // Append 0xFF so that it is larger than value
+        Slice newValue = Slices.copyOf(value, 0, index + 1);
+        newValue.setByte(index, 0xFF);
```

Comment:
Isn't it a problem it won't be utf-8 string afterwards? (If it would, consider something like `Character.MAX_CODE_POINT` instead)